File: csharpServer/server/CryptPassword.cs
namespace server;

public class CryptPassword
{
    public static string HashPassword(string password)
    {
        return BCrypt.Net.BCrypt.HashPassword(password);
    }

    public static bool CheckPassword(string password, string hashedPassword)
    {
        return BCrypt.Net.BCrypt.Verify(password, hashedPassword);
    }
}

File: javaClient/src/main/java/ro/mpp2024/utils/CryptPassword.java
package ro.mpp2024.utils;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;


public class CryptPassword {
    private static final BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();

    public static String hashPassword(String plainPassword) {
        return encoder.encode(plainPassword);
    }

    public static boolean checkPassword(String plainPassword, String hashedPassword) {
        return encoder.matches(plainPassword, hashedPassword);
    }
}
